cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -762,30 +795,37 @@ public QueryHandle createNamespace(Id.Namespace namespace) throws ExploreExcepti
   @Override
   public QueryHandle deleteNamespace(Id.Namespace namespace) throws ExploreException, SQLException {
     startAndWait();
-
-    try {
-      SessionHandle sessionHandle = null;
-      OperationHandle operationHandle = null;
-      Map<String, String> sessionConf = startSession();
-
+    String customHiveDatabase = getCustomHiveDatabaseIfExists(namespace.getId());
+    if (Strings.isNullOrEmpty(customHiveDatabase)) {
       try {
-        sessionHandle = openHiveSession(sessionConf);
+        SessionHandle sessionHandle = null;
+        OperationHandle operationHandle = null;
+        Map<String, String> sessionConf = startSession();
 
-        String database = getHiveDatabase(namespace.getId());
-        String statement = String.format("DROP DATABASE %s", database);
-        operationHandle = doExecute(sessionHandle, statement);
-        QueryHandle handle = saveReadOnlyOperation(operationHandle, sessionHandle, sessionConf, statement, database);
-        LOG.info("Deleting database {} with handle {}", database, handle);
-        return handle;
+        try {
+          sessionHandle = openHiveSession(sessionConf);
+
+          String database = getHiveDatabase(namespace.getId());
+          String statement = String.format("DROP DATABASE %s", database);
+          operationHandle = doExecute(sessionHandle, statement);
+          QueryHandle handle = saveReadOnlyOperation(operationHandle, sessionHandle, sessionConf, statement, database);
+          LOG.info("Deleting database {} with handle {}", database, handle);
+          return handle;
+        } catch (Throwable e) {
+          closeInternal(getQueryHandle(sessionConf),
+                        new ReadOnlyOperationInfo(sessionHandle, operationHandle, sessionConf, "", ""));
+          throw e;
+        }
+      } catch (HiveSQLException e) {
+        throw getSqlException(e);
       } catch (Throwable e) {
-        closeInternal(getQueryHandle(sessionConf),
-                      new ReadOnlyOperationInfo(sessionHandle, operationHandle, sessionConf, "", ""));
-        throw e;
+        throw new ExploreException(e);
       }
-    } catch (HiveSQLException e) {
-      throw getSqlException(e);
-    } catch (Throwable e) {
-      throw new ExploreException(e);
+    } else {
```

Comment:
In case of else we still need to execute a hive query through explore which can raise same exceptions.